File: mobile-engage/src/androidTest/java/com/emarsys/mobileengage/testUtil/DatabaseTestUtils.java
package com.emarsys.mobileengage.testUtil;

import android.support.test.InstrumentationRegistry;

import com.emarsys.core.database.helper.CoreDbHelper;
import com.emarsys.mobileengage.database.MobileEngageDbHelper;

public class DatabaseTestUtils {

    public static final String EMARSYS_MOBILE_ENGAGE_DB = MobileEngageDbHelper.DATABASE_NAME;
    public static final String EMARSYS_CORE_QUEUE_DB = CoreDbHelper.DATABASE_NAME;

    private DatabaseTestUtils() {
    }

    public static void deleteMobileEngageDatabase() {
        InstrumentationRegistry.getContext().deleteDatabase(EMARSYS_MOBILE_ENGAGE_DB);
    }

    public static void deleteCoreDatabase() {
        InstrumentationRegistry.getContext().deleteDatabase(EMARSYS_CORE_QUEUE_DB);
    }
}

File: core/src/androidTest/java/com/emarsys/core/testUtil/DatabaseTestUtils.kt
package com.emarsys.core.testUtil

import android.database.sqlite.SQLiteDatabase
import android.support.test.InstrumentationRegistry

import com.emarsys.core.database.helper.CoreDbHelper

object DatabaseTestUtils {

    val EMARSYS_CORE_QUEUE_DB = CoreDbHelper.DATABASE_NAME

    fun deleteCoreDatabase() {
        InstrumentationRegistry.getContext().deleteDatabase(EMARSYS_CORE_QUEUE_DB)
    }

    fun dropAllTables(db: SQLiteDatabase) {
        db.rawQuery("SELECT 'DROP TABLE ' || name || ';' FROM sqlite_master WHERE type='table';", null).use {
            it.moveToFirst()
            while (!it.isAfterLast) {
                db.execSQL(it.getString(0))
                it.moveToNext()
            }
        }
    }
}